### PR TITLE
docs: add a metadata section

### PR DESCRIPTION
**Pull request by Ulexus**
This adds some text to the Metadata section, briefly describing the
Metadata server and the construction of the Talos configuration.

Signed-off-by: Seán C McCord <ulexus@gmail.com>

### CODE REVIEW

**Comment by Ulexus** [comment]
Fixes #167

File: docs/website/content/docs/v0.1/Configuration/metadata.md [hunk 1]
```diff
@@ -4,3 +4,29 @@ weight: 4
...
+
+  - The `Cluster` of which the `Machine` is a member.
+  - The `ServerClass` which was used to select the `Server` into the `Cluster`.
+  - Any `Server`-specific patches.
+
+The base template is constructed from the Cluster.  Then, any configuration
+patches are applied from the `ServerClass` and `Server`.
+
+Only configuration patches are allowed in the `ServerClass` and `Server`
+resources.  These patches take the form of an [RFC 6902](https://tools.ietf.org/html/rfc6902) JSON (or YAML) patch.
```
Comment (rsmitty) [inline docs/website/content/docs/v0.1/Configuration/metadata.md:23]: May want to point [here](https://www.sidero.dev/docs/v0.1/configuration/servers) with a note saying you can see an example.

File: docs/website/content/docs/v0.1/Configuration/metadata.md [hunk 2]
```diff
@@ -4,3 +4,29 @@ weight: 4
...
+
+## Talos Machine Configuration
+
+The configuration of each machine is constructed from a number of sources:
+
+  - The `Cluster` of which the `Machine` is a member.
+  - The `ServerClass` which was used to select the `Server` into the `Cluster`.
+  - Any `Server`-specific patches.
+
+The base template is constructed from the Cluster.  Then, any configuration
```
Comment (rsmitty) [inline docs/website/content/docs/v0.1/Configuration/metadata.md:19]: Not _quite_ sure how we want to express this, but the base machine config is constructed by the talos bootstrap provider, based on the information present in the cluster manifests that were applied. From there, the patches bit is correct.

File: docs/website/content/docs/v0.1/Configuration/metadata.md [hunk 3]
```diff
@@ -4,3 +4,29 @@ weight: 4
 ---
 
 # Metadata
+
+The Metadata server manages the Machine metadata.  Critically, this includes the
```
Comment (rsmitty) [inline docs/website/content/docs/v0.1/Configuration/metadata.md:8]:
```suggestion
The Metadata server manages the Machine metadata (or in Talos terms, the "machine config").  Critically, this includes the
```

File: docs/website/content/docs/v0.1/Configuration/metadata.md (hunk 2 above)
Comment (Ulexus) [reply docs/website/content/docs/v0.1/Configuration/metadata.md:19]: I was vague there precisely because I didn't know.  ;)

I'll tune it up a bit.

**Pull request merged**